lib/queries/batchInsert.js

```diff
@@ -1,42 +1,25 @@
-import configurable from 'configurable.js';
 import valueSubQuery from './valueSubQuery';
 import batchParameter from './batchParameter';
+import returningQuery from './returningQuery';
 
-function batchInsert(entities) {
-    const {
-        table,
-        fields,
-    } = this.config;
+export default ({ table, fields, returnFields }) => {
+    const returning = returningQuery(returnFields);
 
-    let { returnFields } = this.config;
+    return (entities) => {
+        if (!entities || entities.length === 0) {
+            throw new Error(`No data for batch inserting ${table} entities.`);
```

Comment:
Not sure we need to throw an error here. Batch inserting nothing should work fine (even if it does nothing).